test/tools/unified-spec-runner/match.ts

```diff
@@ -523,6 +567,51 @@ export function matchesEvents(
   }
 }
 
+export function compareLogs(
+  expected: ExpectedLogMessage[],
+  actual: ExpectedLogMessage[],
+  entities: EntitiesMap
+): void {
+  expect(actual).to.have.lengthOf(expected.length);
+
+  for (const [index, actualLog] of actual.entries()) {
+    const rootPrefix = `expectLogMessages[${index}]`;
+    const expectedLog = expected[index];
+
+    // Check that log levels match
+    expect(actualLog).to.have.deep.property('level', expectedLog.level);
+
+    // Check that components match
+    expect(actualLog).to.have.deep.property('component', expectedLog.component);
+
+    // NOTE: The spec states that if the failureIsRedacted flag is present, we
+    // must assert that a failure occurred.
+    if (expectedLog.failureIsRedacted !== undefined) {
+      if (expectedLog.failureIsRedacted) {
+        // Assert that a failure is present and has been redacted
+        expect(actualLog.data.failure, 'Can only use failureIsRedacted when a failure exists').to
+          .exist;
+        expect(actualLog.data.failure, 'Expected failure to have been redacted').to.deep.equal({});
+      } else {
+        // Assert that a failure is present and has not been redacted
+        expect(actualLog.data.failure, 'Can only use failureIsRedacted when a failure exists').to
+          .exist;
+        expect(
+          actualLog.data.failure,
+          'Expected failure to have not been redacted'
+        ).to.not.deep.equal({});
+      }
+    } else {
+      expect(
```

Comment:
is this correct? per the spec, `failureIsRedacted` is optional and *can* be present only on logs that have a `failure`; so anything with the `failureRedacted` flag will definitely have a `failure`, but anything without that flag could either not have a failure or have a failure where we don't care about whether or not it's redacted

the actual existence of the failure would be verified by the `resultCheck`, no? (since it would compare `actualLog.data.failure` against `expectedLog.data.failure`)